src/components/Courses/Courses.tsx

```diff
@@ -1,92 +1,117 @@
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { useNavigate } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
 
 import { CourseCard } from './components/CourseCard';
 import { EmptyCourseList } from './components/EmptyCourseList';
 import { Button } from 'src/common/Button';
 import { SearchBar } from './components/SearchBar';
-import { CoursesProp } from './Courses.types';
-import { TOKEN_KEY_NAME } from 'src/util/CommonConstant';
+import {
+	PERMISSION_ERROR_TEXT_MESSAGE,
+	TOKEN_KEY_NAME,
+} from 'src/util/CommonConstant';
 import { buildCourseCardData } from 'src/util/CourseUtil';
 import { RouterPath } from 'src/util/RouterPath';
+import { RootState } from 'src/store';
+import { DeleteCourseAction } from 'src/store/course/action';
 
 import './Courses.css';
 
-export const Courses: React.FC<CoursesProp> = (coursesProp) => {
+const ADD_NEW_COURSE_BUTTON_TEXT = 'ADD NEW COURSE';
+const SEARCH_NOTHING_FOUND_MESSAGE = 'Nothing Found';
+
+export const Courses: React.FC = () => {
 	const navigate = useNavigate();
-	const [courses, setCourses] = useState(coursesProp.courses);
+	const dispatch = useDispatch();
+
+	if (
+		localStorage.getItem(TOKEN_KEY_NAME) === '' ||
+		localStorage.getItem(TOKEN_KEY_NAME) === null
+	) {
+		navigate(RouterPath.LOGIN);
+	}
+
+	const courseList = useSelector(
+		(state: RootState) => state.courseReducer.courses
+	);
+
 	const [searchValue, setSearchValue] = useState('');
-	const [filteredList, setFilteredList] = useState(coursesProp.courses);
+	const [filteredList, setFilteredList] = useState(courseList);
+
+	useEffect(() => {
+		setFilteredList(courseList);
+	});
 
 	const searchValueOnChange = (event) => {
 		setSearchValue(event.target.value);
 	};
 
 	const search = () => {
-		const filteredList = courses.filter(
+		const filteredList = courseList.filter(
 			(course) =>
 				course.id.toUpperCase().includes(searchValue.toUpperCase()) ||
 				course.title.toUpperCase().includes(searchValue.toUpperCase())
 		);
-		setFilteredList(filteredList);
+		if (filteredList.length > 0) {
```

Comment:
you can omit this check, just use:
```
if (filteredList.length) {}
```
because if filteredList.length === 0 this condition will return false